File: Ej2_Java/src/main/java/org/example/InventarioVideojuegos.java
package org.example;

import java.util.ArrayList;

public class InventarioVideojuegos {

    ArrayList<Videojuego> videojuegos = new ArrayList<>();

    public void addVideojuego(Videojuego videojuego) {
        if (!videojuegos.contains(videojuego)) {
            videojuegos.add(videojuego);
            System.out.println("Videojuego " + videojuego.titulo + " añadido correctamente");
        }else{
            System.out.println("El Videojuego " + videojuego.titulo + " ya existe en la lista");
        }
    }

    public void removeVideojuego(Videojuego videojuego) {
        if (videojuegos.contains(videojuego)) {
            videojuegos.remove(videojuego);
            System.out.println("Videojuego " + videojuego.titulo + " eliminado del inventario correctamente");
        }else{
            System.out.println("El Videojuego " + videojuego.titulo + " no existe en la lista");
        }
    }

    public void mostrarVideojuegos() {
        if (videojuegos.size() > 0) {
            System.out.println("Listado de videojuegos: ");
            videojuegos.forEach(videojuego -> {
                videojuego.mostrarInfo();
            });
        }else{
            System.out.println("No hay videojuegos que mostrar");
        }

    }
}

File: Ej2_Kotlin/src/main/kotlin/InventarioVideojuegos.kt
package org.example

class InventarioVideojuegos {
    private val videojuegos = mutableListOf<Videojuego>()

    fun addJuegos(juego: Videojuego) {
        if (videojuegos.contains(juego)) {
            println("Ya existe este juego")
        }else{
            videojuegos.add(juego)
            println("Juego añadido(${juego.titulo})")
        }
    }

    fun deleteJuegos(juego: Videojuego) {
        if (videojuegos.contains(juego)) {
            videojuegos.remove(juego)
            println("Juego eliminado correctamente(${juego.titulo})")
        }else{
            println("No existe este juego")
        }
    }

    fun mostrarJuegos(){
        if (videojuegos.size > 0) {
            println("Listado de videojuegos:")
            videojuegos.forEachIndexed() { index, juego ->
                println("${index+1} - $juego")
            }
        }else{
            println("No hay videojuegos en la lista")
        }

    }
}